tests/block_tools.py

```diff
@@ -1372,3 +1385,420 @@ def get_full_block_and_block_record(
     )
 
     return full_block, block_record
+
+
+def get_name_puzzle_conditions_test(generator: BlockGenerator, max_cost: int) -> NPCResult:
```

Comment:
This is a test version copy of `get_name_puzzle_conditions()`, it does not do *any* validation. This is important to allow tests to create invalid conditions and spend-bundles. The functions below, `create_test_foliage()` and `create_test_unfinished_block()` are also copies that use the no-validation versions of functions, to allow tests to create invalid blocks/spend bundles.

Reply:
note that all of `block_tools.py` is just test helper functions